user_interface/book_ui.rb

```diff
@@ -0,0 +1,82 @@
+require_relative '../modules/book'
+require_relative '../modules/item'
+require_relative '../modules/label'
+require 'json'
+
+class BookUserInterface
+  FILE_LOCATIONS = {
+    books: './data/books.json',
+    labels: './data/labels.json'
+  }.freeze
+
+  def initialize
+    @books = load_data(:books)
+    @labels = load_data(:labels)
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts 'No published books found.'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Publisher: #{book['publisher']}, Cover State: #{book['cover_state']}"
+      end
+    end
+  end
+
+  def add_a_book
+    book = create_book
+    @books << book.to_json
+    label = create_label
+    @labels << label.to_json
+    label.add_items(@labels)
+    save_data
+  end
+
+  def list_all_labels
+    if @labels.empty?
+      puts 'No labels found.'
+    else
+      @labels.each_with_index do |label, index|
+        puts "#{index + 1}) Title: #{label['title']}, Color: #{label['color']}"
+      end
+    end
+  end
+
+  private
+
+  def load_data(type)
+    path = FILE_LOCATIONS[type]
+    if File.zero?(path)
+      []
+    else
+      JSON.parse(File.read(path))
+    end
+  end
+
+  def save_data
+    FILE_LOCATIONS.each do |type, path|
+      data = instance_variable_get("@#{type}")
+      File.write(path, JSON.pretty_generate(data))
+    end
+  end
```

Comment:
- [ ] it would be better if the app doesn't crash when the `json` file is missing you should  handle that case and create a json file if the file is missing 
<img width="1123" alt="Screenshot 2023-02-17 at 16 21 39" src="https://user-images.githubusercontent.com/61790587/219695603-dce95c23-659a-4aef-a212-229c83ccfb18.png">